packages/remix-server-runtime/jsonify.ts

```diff
@@ -146,8 +146,15 @@ type _tests = [
   // toJson
   Expect<Equal<Jsonify<{ toJSON(): "stuff" }>, "stuff">>,
   Expect<Equal<Jsonify<Date>, string>>,
-  Expect<Equal<Jsonify<{ toJSON(): undefined }>, unknown>>,
-  Expect<Equal<Jsonify<{ toJSON(): Date }>, unknown>>,
+  Expect<Equal<Jsonify<{ toJSON(): any }>, any>>,
+  Expect<Equal<Jsonify<{ toJSON(): undefined }>, never>>,
+  Expect<Equal<Jsonify<{ toJSON(): Date }>, string>>,
```

Comment:
Haven't checked the other test cases, but this one is incorrect.

```js
let result = JSON.parse(JSON.stringify({ toJSON: () => new Date() }))
// => {}
```

So the result is an empty object, not a `string`. `toJSON` doesn't seem to always recursively serialize the return value.

Reply:
I was checking, which did respond with a string

```
let result = JSON.parse(JSON.stringify({ toJSON: () => Date() }))
// => 'Mon Oct 23 2023 03:45:02 GMT+0200 (Midden-Europese zomertijd)'
```